ez/ez.go

```diff
@@ -16,6 +16,33 @@ import (
 	"github.com/vimeo/dials/yaml"
 )
 
+// Option sets an option on dialsOptions
+type Option func(*dialsOptions)
+
+type dialsOptions struct {
+	watch      bool
+	flagConfig *flag.NameConfig
+}
+
+func getDefaultOption() *dialsOptions {
+	return &dialsOptions{
+		flagConfig: flag.DefaultFlagNameConfig(),
+	}
+}
+
+// WithFlagConfig sets the flag NameConfig to the specified one
+func WithFlagConfig(flagConfig *flag.NameConfig) Option {
+	return func(d *dialsOptions) { d.flagConfig = flagConfig }
+}
+
+// WithWatchingConfigFile allows to watch the config file by using the watching
+// file source
+func WithWatchingConfigFile() Option {
+	return func(d *dialsOptions) { d.watch = true }
+}
+
+type decoderFactory func(string) dials.Decoder
```

Comment:
This type should be exported with a docstring, since it's part of the public interface.

Reply:
Exported DecoderFactory and added a doc string